packages/exo/src/exo-tools.js

```diff
@@ -439,12 +426,22 @@ export const defendPrototype = (
     );
   }
 
-  return Far(
-    tag,
-    /** @type {T & import('./get-interface.js').GetInterfaceGuard<T>} */ (
-      prototype
-    ),
-  );
+  if (interfaceGuard) {
+    const getInterfaceGuardMethod = {
+      [GET_INTERFACE_GUARD]() {
+        return interfaceGuard;
+      },
+    }[GET_INTERFACE_GUARD];
+    prototype[GET_INTERFACE_GUARD] = bindMethod(
+      `In ${q(GET_INTERFACE_GUARD)} method of (${tag})`,
+      contextProvider,
+      getInterfaceGuardMethod,
+      thisfulMethods,
+      undefined,
+    );
+  }
+
+  return Far(tag, /** @type {T} */ (prototype));
```

Comment:
Is there a particular reason to make the presence of the `GET_INTERFACE_GUARD` method conditional?  Shouldn't it return the actual interface used in defendPrototype rather than the user-supplied value?

I ask because omitting `GET_INTERFACE_GUARD` instead of always defining it and making it return *something* makes it hard for remote callers to use.  They currently have no way to sniff the presence or absence of the method on a remote object.

I would argue that the original typing of this function is the appropriate one, and would encourage you to find a way to satisfy it instead of using a (sometimes) inaccurate return type.

Reply:
I should clarify:
> the original typing of this function (defendPrototype) is the appropriate one

What I mean is to keep the deleted `return Far(.... /** @type {T & import('./get-interface.js').GetInterfaceGuard<T>} 
 */ ...);` intact, and define `GET_INTERFACE_GUARD` before it, unconditionally, so that no remote-sniffing needs to happen in order to invoke it.  Then, for bonus points (or at least worth a TODO), change `GET_INTERFACE_GUARD`'s return value always to be something meaningful.

Reply:
> instead of always defining it and making it return something makes it hard for remote callers to use

@michaelfig 
I agree this would be an improvement. However, I started on it, and this improvement is a more significant refactor than I anticipated. (Either that, or I missed how to do it simply.) Could we move ahead with this PR without this improvement, together with my filing an issue, and then revisit this improvement in a later PR?

> actual interface ... than the user-supplied value

Currently, there is no actual interface distinct from the user supplied value. The refactor would be to always build it, including in the thisless case where there isn't really even an implicit an interface guard.

Even the improvement as stated doesn't quite obviate the sniffing, since simple `Far` objects would still not have a `GET_INTERFACE_GUARD` method. But addressing that as well is seems impossible without grossly violating our layering.

> keep the deleted ...

That part is now done.

Reply:
https://github.com/endojs/endo/issues/1844

I assigned it to both of us.

Reply:
It just occurred to me that a more pleasant interim solution while waiting for #1844 is to simply have GET_INTERFACE_METHOD() return undefined in that cases where it currently is absent. I just made this change. It still doesn't cover the `Far` case, but it does somewhat cover all the other exo/kind cases by at least making it clear that it isn't yet helping with further sniffing.

Reply:
It does occur to me that if we are content with the possibility of GET_INTERFACE_GUARD() permanently being allowed to return `undefined`, then it would not be a horrible layering violation to have `Far` add only this. Please advise.

Reply:
> `GET_INTERFACE_GUARD()` permanently being allowed to return `undefined`, [and] have `Far` add only this.

💯 I really like this option.  Brilliant!

Reply:
I think you'll like my new variation even better: Have `Far` introduce a `GET_METHOD_NAMES` meta method that applies to all objects made with `Far` or any object they inherit from. This is only a miranda method -- it is only installed if the user does not provide one.

Then, at the cost of a round trip, you can use that to determine whether there is a method named `GET_INTERFACE_GUARD`. Thus, I was able to restore the conditional. I did go further and add a further conjunct to ensure that there is not already a `GET_INTERFACE_GUARD` method, so it is also a miranda method.

Given this, I propose that it is now the correct decision for `GET_INTERFACE_GUARD` to return the interface guard the user provided, since the user can now combine it with the result of `GET_METHOD_NAMES` to calculate the effective interface guard.

PTAL

Reply:
I did make that variation as a separate `feat(pass-style)` commit, so we could back out.

Reply:
This new feature is somewhat tested by the tests I had to modify. I'll wait on writing more tests pending your feedback on this approach.

Reply:
Tests done.